### PR TITLE
MM-70042: Improve project handling in the GitLab integration

### PR DESCRIPTION
**Pull request by edgarbellot**
See [MM-70045](https://mattermost.atlassian.net/browse/MM-70045) for more details.

```release-note
Fixed an issue with project handling in the GitLab integration.
```

## Change Impact: 🟡 Medium

**Reasoning:** The change adds extra namespace/group gating to GitLab project retrieval and permalink preview generation, which directly affects user-facing project access and code preview behavior across multiple components. Existing automated tests cover the primary in-group/out-of-group/unconfigured scenarios, but there’s still moderate risk of blocking or altering edge-case namespaces.

**Regression Risk:** Moderate—behavior changes occur in core GitLab integration logic (project fetching and permalink processing), and failures in the new validation could impact previews even when the initial fetch succeeds.

**QA Recommendation:** Perform focused manual QA for in-group vs out-of-group vs unconfigured GitlabGroup permutations, especially around permalink preview expansion. Skipping manual QA is not recommended due to the user-facing behavioral change, though automated tests should catch many regressions.

*Generated by CodeRabbitAI*

[MM-70045]: https://mattermost.atlassian.net/browse/MM-70045?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/mcp_test.go [hunk 1]
```diff
@@ -394,6 +398,69 @@ func TestHandleAddComment_Validation(t *testing.T) {
...
+		h := http.NewServeMux()
+		h.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+			if req.Method == http.MethodGet && req.URL.Path == "/api/v4/projects/"+projectPath {
+				_ = json.NewEncoder(w).Encode(map[string]interface{}{
+					"id":                  4242,
+					"name":                "repo",
+					"path_with_namespace": projectPath,
+					"visibility":          "private",
+					"default_branch":      "main",
+				})
```
[the remark this answers is not in the document]
Comment (edgarbellot) [reply server/mcp_test.go]: Applied in 90d704f — updated the map literal to use `any` so the file passes gofmt with the repo's rewrite rule (interface{} → any). The lint check is now green.

**Review by nang2049 — APPROVED** [review]
LGTM!

**Comment by avasconcelos114** [comment]
(Sorry for the noise, testing a cherry-pick packport automation)

**Comment by avasconcelos114** [comment]
🤖 Starting cherry-pick of #690 onto the following branches:
- release-1.12
- release-1.13

Results will be posted here when complete.

**Comment by avasconcelos114** [comment]
🤖 Cherry-pick of #690 complete.

- ✅ release-1.12: https://github.com/mattermost/mattermost-plugin-gitlab/pull/698
- ✅ release-1.13: https://github.com/mattermost/mattermost-plugin-gitlab/pull/699
